albumentations/core/composition.py

```diff
@@ -272,13 +261,12 @@ def __call__(self, *args: Any, force_apply: bool = False, **data: Any) -> Dict[s
             data[self.save_key] = OrderedDict()
 
         need_to_run = force_apply or random.random() < self.p
-        if not need_to_run and not self._always_apply:
```

Comment:
I think this should be reverted. I have RandomCrop with always_apply in Compose that has a probability of 0.9. Previously, the crop was always being performed but now it would only be performed 90% of the time.

Reply:
Can you provide an example of the pipeline where `always_apply` would be useful?

Reply:
I have a dataset with images having different shapes. I use RandomCrop with `always_apply` in `Compose` so that the final images after transformations are always the same shape. Without `always_apply`, images in a batch would have different shapes and the batch formation would fail.

Reply:
Will RandomCrop with p=1 do the job?

Reply:
Ideally it should but with the current implementation it doesn't. There is no function currently that checks if any transforms inside Compose has p=1.0. This function would be similar to _always_apply but also checks for transforms that has p=1.0.

Reply:
I do not get it.

```python
transform = A.Compose([A.RandomCrop(p=1), other transforms], p=1)
```

will do the job?

If not, could you please provide minimal example of the pipeline that shows the issue

Reply:
This wouldn't work: `transform = A.Compose([A.RandomCrop(p=1), other transforms], p=0.9)`

Reply:
Let's close this conversation for now and continue the conversation when you could provide code that could not be executed with p=1

Reply:
The transformation above is the code that wouldn't work now. I am mostly concerned about the behaviour change that this PR silently brings. In older version, `A.Compose([A.RandomCrop(p=1, always_apply=True), other transforms], p=0.9)` would always apply RandomCrop but with this PR, RandomCrop would only be applied with 0.9 probability with the same transform.